db/migrate/20170301192129_create_provider_tags_table.rb

```diff
@@ -0,0 +1,13 @@
+class CreateProviderTagsTable < ActiveRecord::Migration[5.0]
+  def change
+    create_table :provider_tags do |t|
```

Comment:
Even though AWS doesn't have a specific ID for tags, this table should still have an `ems_ref` field.  For AWS, to keep the `ems_ref` value unique for the provider, it would have to be populated with something like `"#{resource.ems_ref}_#{key}"`.

Skipping `ems_ref` now will definitely hurt us in the future.  Or, at least, history has shown that it eventually always hurts later.